db/seeds/source_types.rb

```diff
@@ -8,15 +8,29 @@ def update_or_create(attributes)
 end
 
 openshift_json_schema = {
-  :title  => "Configure OpenShift",
-  :fields => [
-    {:component => "text-field", :name => "endpoint.role", :type => "hidden", :initialValue => "kubernetes"},
-    {:component => "text-field", :name => "authentication.authtype", :type => "hidden", :initialValue => "token"},
-    {:component => "text-field", :name => "url", :label => "URL", :validate => [{:type => "url-validator"}]},
-    {:component => "checkbox", :name => "endpoint.verify_ssl", :label => "Verify SSL"},
-    {:component => "text-field", :name => "endpoint.certificate_authority", :label => "Certificate Authority", :condition => {:when => "endpoint.verify_ssl", :is => true}},
-    {:component => "text-field", :name => "authentication.password", :label => "Token", :type => "password"}
-  ]
+  :authentication => {
+    :token => {
+      :meta   => {
+        :name => "Token",
+      },
+      :schema => {
+        :title  => "Configure OpenShift",
+        :fields => [
+          {:component => "text-field", :name => "authentication.authtype", :type => "hidden", :initialValue => "token"},
+          {:component => "text-field", :name => "authentication.password", :label => "Token", :type => "password"}
+        ]
+      }
+    }
+  },
+  :endpoint       => {
+    :title  => "Configure OpenShift endpoint",
+    :fields => [
```

Comment:
is it possible to have similar path for both models? I mean specifically `fields`:
**authentication**: 
* `/<type>/:schema/:fields`

**endpoint**:
* `/fields`

instead -> `/:schema/:fields` for both

Reply:
Actually, I am removing the `:schema` from authentication right now, as it's not useful.

Reply:
I have put the format into the description, what do you think?

Reply:
nice :+1: 
I see 2nd option in authentication, instead of 
```
:name_of_the_authentication_type => {}, 
:name_of_the_authentication_type => {},
```
have an array: 
`[ { :type => ... }, { :type => ... } ]`
I think it's more formal description, but your way is probably easier to implement.
@agrare what do you think?

Reply:
@slemrmartin yeah it does look a little strange having seemingly an arbitrary key there.  I assume @rvsia is getting the key to use from the authentication_type schema but still I'd agree that an array of types looks better

Reply:
Yeah, with keys it is slightly simpler to use, but I have no problem with changing it.